code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in input)
+		total_fuel_joules += F.get_fuel_joules(FALSE)
+
+	return total_fuel_joules
+
+/obj/machinery/ftl_shunt/core/proc/get_fuel_maximum(var/list/input)
+	var/max_fuel_joules = 0
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in input)
+		max_fuel_joules += F.get_fuel_joules(TRUE)
+
+	return max_fuel_joules
+
+obj/machinery/ftl_shunt/core/proc/fuelpercentage()
+	if(!length(fuel_ports))
+		return 0
+	var/fuel_max = get_fuel_maximum(fuel_ports)
+	if(fuel_max == 0)
+		return 0
+	return round(100.0*get_fuel(fuel_ports)/fuel_max, 0.1)
+
+
+/obj/machinery/ftl_shunt/core/proc/use_fuel(var/joules_req)
+	var/joules_per_port
+	var/avail_fuel = get_fuel(fuel_ports)
+	var/list/fueled_ports = list()
+	var/ports_used
+
+	if(joules_req > avail_fuel) //Not enough fuel in the system.
+		return FALSE
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fuel_ports)
+		if(F.has_fuel())
+			fueled_ports += F
+
+	joules_per_port = (joules_req / length(fueled_ports))
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fueled_ports)
+		if(F.use_fuel_joules(joules_per_port))
+			ports_used++
+
+	if(ports_used == length(fueled_ports))
+		return TRUE
+
+/obj/machinery/ftl_shunt/core/proc/get_charge_time()
+	return (ftl_computer.linked.vessel_mass * CHARGE_TIME_PER_TON) * time_multiplier
+
+/obj/machinery/ftl_shunt/core/Process()
+	if((stat & (BROKEN|NOPOWER)) & charging)
+		cancel_shunt()
+
+	if(charging)
+		chargepercent = round(100.0*(world.time - charge_started)/charge_interval, 0.1)
+		if(world.time >= charge_time) //We've probably finished charging up.
+			charging = FALSE
+			if(use_fuel(required_fuel_joules))
+				execute_shunt()
+			else
+				cancel_shunt() //Not enough fuel for whatever reason. Cancel.
+		SSradiation.radiate(src, (active_power_usage / 1000))
+
+/obj/machinery/ftl_shunt/fuel_port
+	name = "superluminal shunt fuel port"
+	desc = "A fuel port for an FTL shunt."
+	icon_state = "empty"
+
+	var/list/global/fuels = list(/decl/material/gas/hydrogen/tritium = 25000, /decl/material/gas/hydrogen/deuterium = 25000, /decl/material/gas/hydrogen = 25000, /decl/material/solid/exotic_matter = 50000)
+	var/obj/item/fuel_assembly/fuel
+	var/obj/machinery/ftl_shunt/core/master
+	var/max_fuel = 0
+
+/obj/machinery/ftl_shunt/fuel_port/on_update_icon()
+	if(fuel)
+		icon_state = "full"
+	else
+		icon_state = "empty"
+
+/obj/machinery/ftl_shunt/fuel_port/Initialize()
+	set_extension(src, /datum/extension/local_network_member)
+	if(initial_id_tag)
+		var/datum/extension/local_network_member/local_network = get_extension(src, /datum/extension/local_network_member)
+		local_network.set_tag(null, initial_id_tag)
+	. = ..()
+
+/obj/machinery/ftl_shunt/fuel_port/Destroy()
+	. = ..()
+	if(master)
+		master.fuel_ports -= src
+	master = null
+	QDEL_NULL(fuel)
+
+/obj/machinery/ftl_shunt/fuel_port/attackby(var/obj/item/O, var/mob/user)
+	if(istype(O, /obj/item/fuel_assembly))
+		if(!fuel)
+			if(!do_after(user, 20, src))
```

Comment:
This is a nitpick, but `2 SECONDS` reads nicer - there's a bunch of other times in your code that could also be turned into `SECONDS`.